### PR TITLE
[Keymap] update gunp layout for Planck

### PR DESCRIPTION
**Pull request by gunpinyo**
Major update on gunp layout for Planck

## Description

This PR contains many updates to my keymap but the main mechanism of binary switching is the same. I also clean up my `readme.md`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by gunpinyo** [comment]
Hi @yanfali, could you merge my PR?
It has been almost 3 years since our last conversation.
I have fixed all of the problems (I believe).

**Review by drashna — APPROVED** [review]

File: keyboards/planck/keymaps/gunp/rules.mk [hunk 1]
```diff
@@ -2,19 +2,27 @@
...
+MOUSEKEY_ENABLE  = yes       # Mouse keys(+4700)
+EXTRAKEY_ENABLE  = yes       # Audio control and System control(+450)
+CONSOLE_ENABLE   = yes       # Console for debug(+400)
+COMMAND_ENABLE   = no        # Commands for debug and configuration
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+BLUETOOTH_ENABLE = no        # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE  = no        # Enable WS2812 RGB underlight.
+                             #   note: this option is not for Planck `light`
+NKRO_ENABLE      = yes       # Nkey Rollover - if this doesn't work, see here:
+# https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (fauxpark) [inline keyboards/planck/keymaps/gunp/rules.mk:15]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = yes          # Audio output
```

File: keyboards/planck/keymaps/gunp/rules.mk [hunk 2]
```diff
@@ -2,19 +2,27 @@
...
+KEY_LOCK_ENABLE      = yes
+DYNAMIC_MACRO_ENABLE = no    # TODO: fix this later
 
+# # for `rev4`
+# AUDIO_ENABLE         = no    # Audio output on port C6
+# DEBOUNCE_TYPE        = sym_defer_g
+
+# for `light` and `rev6`
+AUDIO_ENABLE         = yes   # Audio output on port C6
+DEBOUNCE_TYPE        = sym_defer_pk
```
Comment (fauxpark) [inline keyboards/planck/keymaps/gunp/rules.mk:28]:
```suggestion
KEY_LOCK_ENABLE = yes
DEBOUNCE_TYPE = sym_defer_pk
```

File: keyboards/planck/keymaps/gunp/rules.mk [hunk 3]
```diff
@@ -2,19 +2,27 @@
 #   change to "no" to disable the options, or define them in the Makefile in
 #   the appropriate keymap folder that will get included automatically
 #
```
Comment (fauxpark) [inline keyboards/planck/keymaps/gunp/rules.mk:4]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please also undo the permissions changes.

**Comment by gunpinyo** [comment]
@fauxpark, all fixed.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**